mediapipe/graphs/instantmotiontracking/calculators/gl_animation_overlay_calculator.cc

```diff
@@ -626,36 +635,45 @@ ::mediapipe::Status GlAnimationOverlayCalculator::GlSetup() {
 
     // vertex position in threespace
     attribute vec4 position;
+    attribute vec3 normal;
 
     // texture coordinate for each vertex in normalized texture space (0..1)
     attribute mediump vec4 texture_coordinate;
 
     // texture coordinate for fragment shader (will be interpolated)
-    varying mediump vec2 sample_coordinate;
+    varying mediump vec3 sample_coordinate;
+    varying mediump vec3 vNormal;
 
     void main() {
-      sample_coordinate = texture_coordinate.xy;
+      sample_coordinate = texture_coordinate.xyz;
       mat4 mvpMatrix = perspectiveMatrix * modelMatrix;
       gl_Position = mvpMatrix * position;
+
+      vec4 tmpNormal = mvpMatrix * vec4(normal, 1.0);
+      vNormal = normalize(tmpNormal.xyz);
     }
   )";
 
   const GLchar *frag_src = R"(
     precision mediump float;
 
-    varying vec2 sample_coordinate;  // texture coordinate (0..1)
+    varying vec3 sample_coordinate;  // texture coordinate (0..1)
+    varying vec3 vNormal;
     uniform sampler2D texture;  // texture to shade with
 
+    // Define ambient lighting constants
+    const vec3 ambient = vec3(-0.1, -0.1, -0.1);
+
     void main() {
       // Sample the texture, retrieving an rgba pixel value
-      vec4 pixel = texture2D(texture, sample_coordinate);
-
+      vec4 pixel = texture2D(texture, sample_coordinate.xy);
       // If the alpha (background) value is near transparent, then discard the
       // pixel, this allows the rendering of transparent background GIFs
       if (pixel.a < 0.2) discard;
+      // Convert rgba to rgb
+      vec3 pixelRGB = pixel.rgb;
 
-      // Set the fragment color to the transformed pixel
-      gl_FragColor = pixel;
+      gl_FragColor = vec4(pixelRGB + ambient, 1.0);
```

Comment:
nit: this is small enough that unless we plan on doing more operations to it, we can just put pixel.rgb here and simplify the code a little